test/e2e/case9_templates_test.go

```diff
@@ -282,3 +285,150 @@ var _ = Describe("Test encrypted policy templates", func() {
+				replicatedPlc.SetAnnotations(annotations)
+				_, err = clientHubDynamic.Resource(gvrPolicy).Namespace(managedCluster).Update(
+					context.TODO(), replicatedPlc, metav1.UpdateOptions{},
+				)
+				Expect(err).To(BeNil())
+
+				By("Verifying the replicated policy against a snapshot")
+				yamlPlc := utils.ParseYaml(case9PolicyYamlCopiedRepl + managedCluster + ".yaml")
+				Eventually(func() interface{} {
+					replicatedPlc = utils.GetWithTimeout(
+						clientHubDynamic,
+						gvrPolicy,
+						testNamespace+"."+case9PolicyNameCopy,
+						managedCluster,
+						true,
+						defaultTimeoutSeconds,
+					)
+
+					return replicatedPlc.Object["spec"]
+				}, defaultTimeoutSeconds, 1).Should(utils.SemanticEqual(yamlPlc.Object["spec"]))
+			})
+
+			It("should reconcile when the secret referenced in the template is updated", func() {
+				By("Updating the secret " + case9SecretName)
+				newToken := "THVrZS4gSSBhbSB5b3VyIGZhdGhlci4="
+				patch := []byte(`{"data": {"token": "` + newToken + `"}}`)
+				_, err := clientHub.CoreV1().Secrets(testNamespace).Patch(
+					context.TODO(), case9SecretName, types.StrategicMergePatchType, patch, metav1.PatchOptions{},
+				)
+				Expect(err).To(BeNil())
+
+				By("Verifying the replicated policy was updated")
+				expected := "$ocm_encrypted:dbHPzG98PxV7RXcAx25mMGPBAUbfjJTEMyFc7kE2W7U3FW5+X31LkidHu/25ic4m"
+				Eventually(func() string {
+					replicatedPlc := utils.GetWithTimeout(
+						clientHubDynamic,
+						gvrPolicy,
+						testNamespace+"."+case9PolicyNameCopy,
+						managedCluster,
+						true,
+						defaultTimeoutSeconds,
+					)
+
+					templates, _, _ := unstructured.NestedSlice(replicatedPlc.Object, "spec", "policy-templates")
+					if len(templates) < 1 {
+						return ""
+					}
+
+					template, ok := templates[0].(map[string]interface{})
+					if !ok {
+						return ""
+					}
+
+					objectTemplates, _, _ := unstructured.NestedSlice(
+						template, "objectDefinition", "spec", "object-templates",
+					)
+					if len(objectTemplates) < 1 {
+						return ""
+					}
+
+					objectTemplate, ok := objectTemplates[0].(map[string]interface{})
+					if !ok {
+						return ""
+					}
+
+					secretValue, _, _ := unstructured.NestedString(
+						objectTemplate, "objectDefinition", "data", "token",
+					)
+
+					return secretValue
+				}, defaultTimeoutSeconds, 1).Should(Equal(expected))
+			})
+
+			It("should clean up the encryption key", func() {
+				utils.Kubectl("delete", "secret",
+					case9EncryptionSecretName,
+					"-n", managedCluster)
+				utils.GetWithTimeout(
+					clientHubDynamic, gvrSecret, case9EncryptionSecretName, managedCluster,
+					false, defaultTimeoutSeconds,
+				)
+			})
+
+			It("should clean up", func() {
+				utils.Kubectl("delete", "-f", case9PolicyYamlCopy, "-n", testNamespace)
+				opt := metav1.ListOptions{}
+				utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 0, false, defaultTimeoutSeconds)
+			})
```

Comment:
The way you have the test organized (which is otherwise ok), I don't think these can *just* be in an AfterAll, but it would be nice to have an extra `AfterAll` version of them (ignoring not-found errors) to make sure things are cleaned up even on failures.

Reply:
I added an AfterAll which I think handles this request.  Thanks!